Field invoked, but object selected

nicot opened the pull request:
This PR fixes the case where a user would select an object with a field invoked on it, and get the error "method or field not found", when they should find the type of the selected object.

nicot committed:
Case field invoked, but object selected

nicot committed:
fixup! Case field invoked, but object selected

sqs commented on godefinfo_test.go:
```diff
@@ -100,6 +100,7 @@ func init() {
 
 	z := T{} //z: p T
 	z // p T
+    z.M0 //z: p T
```
Incorrect indentation wrt the above line

sqs commented:
LGTM after that 1 comment if tests pass.

nicot committed:
fixup! fixup! Case field invoked, but object selected

nicot commented:
```
nico@Dominics-MacBook-Pro> go install && go test                                                                                                   ~/go/src/github.com/sqs/godefinfo
PASS
ok      github.com/sqs/godefinfo    1.071s
```

Good to go.

nicot committed:
Struct literal context

This commit adds context for struct literals. Now you can see the
defInfo page of that field of the struct, instead of the underlying
type.

nicot commented:
I added some more testing and features for struct literals.
